### PR TITLE
test: raise_test_coverage_to_100%

### PR DESCRIPTION
**Pull request by saladgg**

### CODE REVIEW

File: apps/dashboards/tests/test_api.py [hunk 1]
```diff
@@ -92,3 +96,33 @@ def test_delete(self):
         url = reverse("dashboard-detail", kwargs={"pk": fake_data.id})
         response = self.client.delete(url)
         assert response.status_code == 204
+
+
+class Test_Dashboard_and_VisualizationViewSet_non_Staff_Request(
```
Comment (kennedykori) [inline apps/dashboards/tests/test_api.py:101]: Name is too long, move these tests to their respective viewset test cases. See below for more details.

File: apps/dashboards/tests/test_api.py [hunk 2]
```diff
@@ -92,3 +96,33 @@ def test_delete(self):
...
+    LoggedInMixin, APITestCase
+):
+    def setUp(self):
+        self.factory = APIRequestFactory()
+        super().setUp()
+
+    def test_dashboard_list_view_as_non_staff(self):
```
Comment (kennedykori) [inline apps/dashboards/tests/test_api.py:108]: This belongs to the `TestDashboardViewSet` test case.

File: apps/dashboards/tests/test_api.py [hunk 3]
```diff
@@ -92,3 +96,33 @@ def test_delete(self):
...
+            reverse("dashboard-list"),
+        )
+        request.user = self.user
+        request.user.is_staff = False
+        view = DashboardViewSet.as_view(actions={"get": "list"})
+        force_authenticate(request, user=self.user)
+        response = view(request)
+        assert response.status_code == 200
+
+    def test_visualization_list_view_as_non_staff(self):
```
Comment (kennedykori) [inline apps/dashboards/tests/test_api.py:119]: This belongs to the `TestVisualizationViewSet` test case.

File: apps/frontend/tests/test_context_processors.py [hunk 4]
```diff
@@ -0,0 +1,29 @@
...
+
+from apps.common.tests.test_common import LoggedInMixin
+from apps.frontend.context_processors import dashboards
+
+pytestmark = pytest.mark.django_db
+
+fake = Faker()
+
+
+class InitializeTestData(LoggedInMixin, APITestCase):
```
Comment (kennedykori) [inline apps/frontend/tests/test_context_processors.py:13]: Same as above, use a more descriptive name.

File: apps/frontend/tests/test_views.py [hunk 5]
```diff
@@ -0,0 +1,13 @@
+from unittest import TestCase
+
+from apps.common.tests.test_common import LoggedInMixin
+from apps.frontend.views import HomeView
+
+
+class TestViews(LoggedInMixin, TestCase):
+    def setUp(self):
```
Comment (kennedykori) [inline apps/frontend/tests/test_views.py:8]: You can remove this override if you aren't extending it.

File: apps/frontend/tests/test_views.py [hunk 6]
```diff
@@ -0,0 +1,13 @@
...
+        super().setUp()
+
+    def test_home_view_context(self):
```
Comment (kennedykori) [inline apps/frontend/tests/test_views.py:11]: No assert statements.

File: apps/sql_data/tests/test_admin.py [hunk 7]
```diff
@@ -0,0 +1,40 @@
...
+from apps.common.tests.test_common import LoggedInMixin
+from apps.sql_data.admin import DataSourceVersionAdmin
+from apps.sql_data.models import DataSourceVersion, SQLDatabaseSource
+
+pytestmark = pytest.mark.django_db
+
+fake = Faker()
+
+
+class InitializeTestData(LoggedInMixin, APITestCase):
```
Comment (kennedykori) [inline apps/sql_data/tests/test_admin.py:16]: As stated above, use a more descriptive name and also consider inheriting from a more appropriate base class.

File: apps/sql_data/tests/test_admin.py [hunk 8]
```diff
@@ -0,0 +1,40 @@
...
+    def setUp(self):
+        self.factory = APIRequestFactory()
+        self.data_source_version = baker.make(DataSourceVersion)
+        self.sql_data_source = baker.make(SQLDatabaseSource)
+        super().setUp()
```
Comment (kennedykori) [inline apps/sql_data/tests/test_admin.py:21]: Put the call to `super()` as the first statement.

File: apps/sql_data/tests/test_admin.py [hunk 9]
```diff
@@ -0,0 +1,40 @@
...
+
+    def test_audit_base_model_audit_field_set_property(self):
```
Comment (kennedykori) [inline apps/sql_data/tests/test_admin.py:23]: No assert statements!

File: apps/sql_data/tests/test_admin.py [hunk 10]
```diff
@@ -0,0 +1,40 @@
...
+        admin = DataSourceVersionAdmin(
+            model=DataSourceVersion, admin_site=AdminSite()
+        )
+        return admin.audit_details_fieldset
```
Comment (kennedykori) [inline apps/sql_data/tests/test_admin.py:27]: Test methods shouldn't return anything.

File: apps/sql_data/tests/test_admin.py [hunk 11]
```diff
@@ -0,0 +1,40 @@
...
+
+    def test_save_model(self):
```
Comment (kennedykori) [inline apps/sql_data/tests/test_admin.py:29]: No assert statements!

File: apps/sql_data/tests/test_models.py [hunk 12]
```diff
@@ -35,6 +37,17 @@ def test_upload_metadata(self):
             self.extract_metadata.data_source.name
             == self.sql_upload_metadata.data_source_name
         )
+        assert (
+            self.sql_upload_metadata.chunks_count
+            == self.upload_chunk.upload_metadata.chunks_count
+        )
+        assert self.sql_upload_metadata.mark_as_complete(
```
Comment (kennedykori) [inline apps/sql_data/tests/test_models.py:44]: What's happening here? Do you mind breaking this down into multiple statements to make it clear what's going on.

In fact, this test method is testing to many things. Consider breaking it down into multiple test methods.

File: apps/users/tests/test_adapters.py [hunk 13]
```diff
@@ -0,0 +1,30 @@
...
+
+from apps.common.tests.test_common import LoggedInMixin
+from apps.users.adapters import AccountAdapter, SocialAccountAdapter
+
+pytestmark = pytest.mark.django_db
+
+fake = Faker()
+
+
+class InitializeTestData(LoggedInMixin, APITestCase):
```
Comment (kennedykori) [inline apps/users/tests/test_adapters.py:13]: Use a more descriptive name.

File: apps/users/tests/test_adapters.py [hunk 14]
```diff
@@ -0,0 +1,30 @@
...
+    def setUp(self):
+        self.factory = APIRequestFactory()
+        super().setUp()
```
Comment (kennedykori) [inline apps/users/tests/test_adapters.py:16]: Call to `super()` should be the first statement.

File: apps/users/tests/test_adapters.py [hunk 15]
```diff
@@ -0,0 +1,30 @@
...
+
+    def test_account_adapter(self):
+        request = self.factory.request()
+        request.user = self.user
+        response = AccountAdapter.is_open_for_signup(self, request)
+        assert response
```
Comment (kennedykori) [inline apps/users/tests/test_adapters.py:22]: Add more asserts. And also, if the intention is to check if a value is not `None`, `assert value is not None` is more readable and clearer.

File: apps/users/tests/test_adapters.py [hunk 16]
```diff
@@ -0,0 +1,30 @@
...
+
+    def test_social_account_adapter(self):
+        request = self.factory.request()
+        request.user = self.user
+        response = SocialAccountAdapter.is_open_for_signup(
+            self, request, sociallogin=None
+        )
+        assert response
```
Comment (kennedykori) [inline apps/users/tests/test_adapters.py:30]: Same as above.

File: pyproject.toml [hunk 17]
```diff
@@ -11,7 +11,6 @@ directory = "coverage"
 exclude_lines = [
     # Have to re-enable the standard pragma"
     "pragma: no branch",
-    "pragma: nocover",
```
Comment (kennedykori) [inline pyproject.toml:14]: Why did you remove this?

File: apps/users/tests/test_signals.py [hunk 18]
```diff
@@ -0,0 +1,19 @@
...
+
+from apps.common.tests.test_common import LoggedInMixin
+from apps.users.signals import assign_basic_permissions
+
+pytestmark = pytest.mark.django_db
+
+fake = Faker()
+
+
+class InitializeTestData(LoggedInMixin, APITestCase):
```
Comment (kennedykori) [inline apps/users/tests/test_signals.py:13]: Use a more descriptive name.

File: apps/users/tests/test_signals.py [hunk 19]
```diff
@@ -0,0 +1,19 @@
...
+    def setUp(self):
+        self.factory = APIRequestFactory()
+        super().setUp()
```
Comment (kennedykori) [inline apps/users/tests/test_signals.py:16]: Make this the first statement.

**Review by kennedykori — CHANGES_REQUESTED** [review]

File: apps/core/tests/test_models.py [hunk 20]
```diff
@@ -0,0 +1,30 @@
...
+
+fake = Faker()
+pytestmark = pytest.mark.django_db
+
+
+class AuditBaseModelTest(TestCase):
+    """Test for AuditBase."""
+
+    def setUp(self):
+        self.user = baker.make(settings.AUTH_USER_MODEL, email=fake.email())
```
[the remark this answers is not in the document]
Comment (saladgg) [reply apps/core/tests/test_models.py]: Fixed

File: apps/core/tests/test_models.py [hunk 21]
```diff
@@ -0,0 +1,30 @@
...
+        super().setUp()
+
+    def test_created_and_created_by(self):
```
[the remark this answers is not in the document]
Comment (saladgg) [reply apps/core/tests/test_models.py:21]: Fixed

File: apps/dashboards/tests/test_api.py (hunk 1 above)
Comment (saladgg) [reply apps/dashboards/tests/test_api.py]: Fixed

File: apps/dashboards/tests/test_api.py (hunk 2 above)
Comment (saladgg) [reply apps/dashboards/tests/test_api.py]: Fixed

File: apps/users/tests/test_adapters.py (hunk 15 above)
Comment (saladgg) [reply apps/users/tests/test_adapters.py]: Fixed

File: apps/users/tests/test_adapters.py (hunk 16 above)
Comment (saladgg) [reply apps/users/tests/test_adapters.py]: Fixed

File: pyproject.toml (hunk 17 above)
Comment (saladgg) [reply pyproject.toml:14]: There was two occurences of "pragma: nocover",

File: apps/users/tests/test_signals.py (hunk 18 above)
Comment (saladgg) [reply apps/users/tests/test_signals.py]: Done.

File: apps/users/tests/test_signals.py (hunk 19 above)
Comment (saladgg) [reply apps/users/tests/test_signals.py]: Done

File: apps/core/tests/test_models.py [hunk 22]
```diff
@@ -0,0 +1,30 @@
...
+from model_bakery import baker
+
+from apps.sql_data.models import SQLDatabaseSource
+from apps.users.models import User
+
+fake = Faker()
+pytestmark = pytest.mark.django_db
+
+
+class AuditBaseModelTest(TestCase):
```
[the remark this answers is not in the document]
Comment (saladgg) [reply apps/core/tests/test_models.py:14]: Done.

File: apps/core/tests/test_models.py (hunk 21 above)
[the remark this answers is not in the document]
Comment (kennedykori) [reply apps/core/tests/test_models.py:21]: The assert statement(s) should reflect what is being tested.

Also, testing whether the `pk` property is `None` after a save is beyond the scope of our project. We did not modify or customize that feature so we do not need to test it.

File: apps/dashboards/tests/test_api.py [hunk 23]
```diff
@@ -35,11 +40,22 @@ def test_get(self):
...
+    def test_list_view_as_non_staff(self):
+        request = self.factory.get(
+            reverse("visualization-list"),
+        )
+        request.user = self.user
+        request.user.is_staff = False
+        view = VisualizationViewSet.as_view(actions={"get": "list"})
+        force_authenticate(request, user=self.user)
+        response = view(request)
+        assert response.status_code == 200
```
Comment (kennedykori) [inline apps/dashboards/tests/test_api.py:57]: This is a step in the right direction but I would expect to see more detailed tests here. Add a test that actually shows that a non_staff user cannot access some visualizations or can only access certain visualizations.

Remember that the goal is to actually test the current implementation and not just to reach a coverage of 100%.

File: apps/dashboards/tests/test_api.py [hunk 24]
```diff
@@ -76,11 +93,22 @@ def test_get(self):
...
+    def test_list_view_as_non_staff(self):
+        request = self.factory.get(
+            reverse("dashboard-list"),
+        )
+        request.user = self.user
+        request.user.is_staff = False
+        view = DashboardViewSet.as_view(actions={"get": "list"})
+        force_authenticate(request, user=self.user)
+        response = view(request)
+        assert response.status_code == 200
```
Comment (kennedykori) [inline apps/dashboards/tests/test_api.py:110]: The comment above still applies here.

File: apps/dashboards/tests/test_models.py [hunk 25]
```diff
@@ -0,0 +1,24 @@
...
+pytestmark = pytest.mark.django_db
+
+fake = Faker()
+
+
+class InitializeTestData(LoggedInMixin, APITestCase):
+    def setUp(self):
+        self.dashboard = baker.make(Dashboard)
+        self.visualization = baker.make(Visualization)
+        super().setUp()
```
Comment (kennedykori) [inline apps/dashboards/tests/test_models.py:18]: Make `super().setUp()` the first statement.

File: apps/sql_data/tests/test_api.py [hunk 26]
```diff
@@ -32,6 +32,15 @@ def test_update(self):
...
         assert response.status_code == 200
 
+    def test_start_chunk_upload_fail(self):
+        data = {"bad_data": 4}
+        fake_data = baker.make(SQLUploadChunk)
+        url = reverse(
+            "sqluploadmetadata-start-chunk-upload", kwargs={"pk": fake_data.id}
+        )
+        response = self.client.post(url, data)
+        assert response.status_code == 400
```
Comment (kennedykori) [inline apps/sql_data/tests/test_api.py:42]: No happy case.

File: apps/sql_data/tests/test_models.py [hunk 27]
```diff
@@ -19,14 +19,16 @@
 
 class InitializeTestData(LoggedInMixin, APITestCase):
     def setUp(self):
```
Comment (kennedykori) [inline apps/sql_data/tests/test_models.py:21]: Add more descriptive name for this test class.

File: pyproject.toml (hunk 17 above)
Comment (kennedykori) [reply pyproject.toml:14]: Yes, with different spelling. But it's okay 👍🏿

**Review by kennedykori — CHANGES_REQUESTED** [review]

File: apps/dashboards/tests/test_api.py [hunk 28]
```diff
@@ -45,15 +45,26 @@ def test_list_view(self):
...
 
+    def test_list_view_as_staff(self):
+        request = self.factory.get(
+            reverse("visualization-list"),
+        )
+        request.user = self.user
+        request.user.is_staff = True
+        view = VisualizationViewSet.as_view(actions={"get": "list"})
+        response = view(request)
+        assert response.data["count"] > 0
```
Comment (kennedykori) [inline apps/dashboards/tests/test_api.py:56]: Specify the count. Just greater than zero is a cheat.

File: apps/dashboards/tests/test_api.py [hunk 29]
```diff
@@ -45,15 +45,26 @@ def test_list_view(self):
...
     def test_list_view_as_non_staff(self):
         request = self.factory.get(
             reverse("visualization-list"),
         )
         request.user = self.user
         request.user.is_staff = False
         view = VisualizationViewSet.as_view(actions={"get": "list"})
-        force_authenticate(request, user=self.user)
         response = view(request)
+        assert response.data["count"] >= 0
```
Comment (kennedykori) [inline apps/dashboards/tests/test_api.py:67]: Same as above, specify the count. Show that there is an actual difference between the number of visualizations that a staff user can access as compared to a non- staff user

File: apps/dashboards/tests/test_api.py [hunk 30]
```diff
@@ -98,6 +109,18 @@ def test_list_view(self):
...
+    def test_list_view_as_staff(self):
+        request = self.factory.get(
+            reverse("dashboard-list"),
+        )
+        request.user = self.user
+        request.user.is_staff = True
+        view = DashboardViewSet.as_view(actions={"get": "list"})
+        force_authenticate(request, user=self.user)
+        response = view(request)
+        assert response.data["count"] > 0
```
Comment (kennedykori) [inline apps/dashboards/tests/test_api.py:121]: Same as the visualizations tests, show an actual count.

File: apps/dashboards/tests/test_api.py [hunk 31]
```diff
@@ -107,6 +130,7 @@ def test_list_view_as_non_staff(self):
         view = DashboardViewSet.as_view(actions={"get": "list"})
         force_authenticate(request, user=self.user)
         response = view(request)
+        assert response.data["count"] >= 0
```
Comment (kennedykori) [inline apps/dashboards/tests/test_api.py:133]: Specify a count.

File: apps/dashboards/tests/test_models.py [hunk 32]
```diff
@@ -11,11 +11,11 @@
 fake = Faker()
 
 
-class InitializeTestData(LoggedInMixin, APITestCase):
+class TestDashboardModels(LoggedInMixin, APITestCase):
```
Comment (kennedykori) [inline apps/dashboards/tests/test_models.py:14]: `TestDashboard` will suffice.

File: apps/sql_data/tests/test_admin.py [hunk 33]
```diff
@@ -13,12 +13,12 @@
 fake = Faker()
 
 
-class InitializeTestData(LoggedInMixin, APITestCase):
+class TestCoreModels(LoggedInMixin, APITestCase):
```
Comment (kennedykori) [inline apps/sql_data/tests/test_admin.py:16]: This name is misleading or this test case is in the wrong module. Use a more appropriate name.

**Review by kennedykori — CHANGES_REQUESTED** [review]

File: apps/core/tests/test_models.py [hunk 34]
```diff
@@ -0,0 +1,73 @@
...
+            self.sql_upload_metadata.name
+        )
+
+    def test_is_complete_false(self):
+        self.sql_upload_metadata.finish_time = None
+        assert str(self.sql_upload_metadata.is_complete) is str(False)
+
+    def test_is_complete_true(self):
+        self.sql_upload_metadata.finish_time = datetime.now()
+        assert str(self.sql_upload_metadata.is_complete) is str(True)
```
Comment (kennedykori) [inline apps/core/tests/test_models.py:73]: Fix as discussed

File: apps/sql_data/tests/test_models.py [hunk 35]
```diff
@@ -17,46 +17,64 @@
...
+
+    def test_data_source_name(self):
+        assert str(self.sql_upload_metadata.data_source_name) == str(
+            self.sql_upload_metadata.extract_metadata.data_source.name
         )
+
+    def test_upload_completion(self):
+        assert str(
+            self.sql_upload_metadata.mark_as_complete(self.user)
+        ) == str(None)
```
Comment (kennedykori) [inline apps/sql_data/tests/test_models.py:80]: Fix as discussed

File: config/settings/base.py [hunk 36]
```diff
@@ -72,6 +72,7 @@
     "rest_framework",
     "rest_framework.authtoken",
     "django_filters",
+    "django_extensions",
```
Comment (kennedykori) [inline config/settings/base.py:75]: Order alphabetically as discussed.

**Review by kennedykori — CHANGES_REQUESTED** [review]

**Review by kennedykori — APPROVED** [review]
